### PR TITLE
Support Python 3.7

### PR DESCRIPTION
**Pull request by wenleix**
Google Collab runs Python 3.7

### CODE REVIEW

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ jobs:
       - name: Print CPU info
         run: cat /proc/cpuinfo
 
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
```
Comment (ejguan) [inline .github/workflows/ubuntu.yml:22]: Do you want to use matrix of python versions? We have a reference from torchdata. https://github.com/pytorch/data/blob/2d94ebc6e95d4bd475a98e947781e58410386a10/.github/workflows/ci.yml#L19-L28

**Review by ejguan — APPROVED** [review]
LGTM with a comment about python version matrix

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (wenleix) [reply .github/workflows/ubuntu.yml:22]: @ejguan we should. But the current build/test time is not not ideal so just keep the most important test for now...

File: setup.py [hunk 2]
```diff
@@ -135,20 +135,21 @@ def build_extension(self, ext):
     license="BSD",
     install_requires=[
         "arrow",
-        "numpy",
+        "numpy==1.21.4",
```
Comment (wenleix) [inline setup.py:138]: otherwise, it will pick up `1.22.0rc1` and doesn't find Python 3.7 distribution.

@ejguan  have you seen similar issues before?

**Pull request closed without merging**